### DISCUSSION TITLE
How to focus out from react ReactNodeViewRenderer after addNodeView?

### DISCUSSION THREAD

**Discussion by emewjin** [ueberdosis/tiptap#4451] [General Questions]
I created an inline Custom Node with ReactNodeViewRenderer, as shown in the code below:
```tsx
  inline: true,
  group: 'inline',

  addNodeView() {
    return ReactNodeViewRenderer(CustomNodeView);
  },
```
When a NodeView is added to the document, the cursor is placed after the ReactNodeViewRenderer. 
Therefore, if I write text, the text is inserted next to the ReactNodeViewRenderer, as shown in the image below:
![image](https://github.com/ueberdosis/tiptap/assets/76927618/3d447d1e-93b1-4024-84a8-207fbfbd8dea)

I have already added `isolating: true`, but it didn't work as expected. 
How can I move the cursor out and always create a new <p> tag when I write text after adding the node view?

**Comment by fbolcic** [comment]
I have a similar problem. Did you find any solution for this?

**Reply by emewjin** [reply]
@fbolcic

I couldn’t find the perfect solution, and the issue was resolved by changing the node to a block node.

But, let me tell you what I tried on `inline` node. I'll write as I remember, but my memory may not be exact. I'll ask for your understanding in advance.

> +) The `isolating` option was only applied to `block` nodes.

At first, after inserting a Custom Node, maybe I executed the `createParagraph` function to force the cursor to be positioned in a new Paragraph right after the Custom Node. 
However, I found that I could still move the cursor inside the Custom Node using arrow keys. 
So, I was contemplating how to prevent text input inside the Custom Node.

I created a new `block` Custom Node that would serve as the **parent** of the original Custom Node (hereinafter referred to as the parent) and set it up so that only Custom Nodes could be `content`. 
This way, the cursor would still be positioned inside Custom Node cause it is still `inline` node, but it wouldn’t be possible to input separate text. 
I received help from the prosemirror API documentation. However, this approach was not perfect because there was still a trailing break.
The presence of a trailing break without being able to input anything seemed like a bug.

Unfortunately, I couldn’t find a way to remove the trailing break, so I try to make it invisible by using CSS. However, trailing breaks are almost essential elements in the editor, so this caused other bugs.

Also, when deleting a node using the backspace, only the Custom Node inside the parent was deleted, and the parent remained.

That's why I gave up inline and switched to `block`. If someone knows the exact solution, I hope you can share it here. I've been waiting.

**Reply by fbolcic** [reply]
Thanks for the explanation 👍
